server/channels/store/sqlstore/scheduled_post_store.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"sync"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	sq "github.com/mattermost/squirrel"
+	"github.com/pkg/errors"
+)
+
+type SqlScheduledPostStore struct {
+	*SqlStore
+	maxMessageSizeOnce   sync.Once
+	maxMessageSizeCached int
+}
+
+func newScheduledPostStore(sqlStore *SqlStore) *SqlScheduledPostStore {
+	return &SqlScheduledPostStore{
+		SqlStore:             sqlStore,
+		maxMessageSizeCached: model.PostMessageMaxRunesV2,
+	}
+}
+
+func (s *SqlScheduledPostStore) columns(prefix string) []string {
+	if prefix != "" {
+		prefix = prefix + "."
+	}
+
+	return []string{
+		prefix + "Id",
+		prefix + "CreateAt",
+		prefix + "UpdateAt",
+		prefix + "UserId",
+		prefix + "ChannelId",
+		prefix + "RootId",
+		prefix + "Message",
+		prefix + "Props",
+		prefix + "FileIds",
+		prefix + "Priority",
+		prefix + "ScheduledAt",
+		prefix + "ProcessedAt",
+		prefix + "ErrorCode",
+	}
+}
+
+func (s *SqlScheduledPostStore) scheduledPostToSlice(scheduledPost *model.ScheduledPost) []interface{} {
+	return []interface{}{
+		scheduledPost.Id,
+		scheduledPost.CreateAt,
+		scheduledPost.UpdateAt,
+		scheduledPost.UserId,
+		scheduledPost.ChannelId,
+		scheduledPost.RootId,
+		scheduledPost.Message,
+		model.StringInterfaceToJSON(scheduledPost.GetProps()),
+		model.ArrayToJSON(scheduledPost.FileIds),
+		model.StringInterfaceToJSON(scheduledPost.Priority),
+		scheduledPost.ScheduledAt,
+		scheduledPost.ProcessedAt,
+		scheduledPost.ErrorCode,
+	}
+}
+
+func (s *SqlScheduledPostStore) Save(scheduledPost *model.ScheduledPost) (*model.ScheduledPost, error) {
```

Comment:
Is this going to also be able to `UPDATE` the scheduled posts? If not, we shouldn't use the name `Save` as I think that's better suited to upsert functions.

Reply:
No, update will have a different store method. Updated to be `CreateScheduledPost`.